api/domain/cart.repository.go

```diff
@@ -0,0 +1,11 @@
+package domain
+
+import (
+	"context"
+)
+
+type CartRepository interface {
+	AddArtToyToCart(ctx context.Context, userId int64, artToyId int64) error
+	FindArtToysInCart(ctx context.Context, userId int64) ([]*ArtToy, error)
+	DeleteArtToysInCart(ctx context.Context, userId int64, artToyIds []int64) error
+}
```

Comment:
```suggestion
package domain

import (
	"context"
)

type CartRepository interface {
	AddArtToyToCart(ctx context.Context, userId int64, artToyId int64) error
	FindArtToysInCart(ctx context.Context, userId int64) ([]*ArtToy, error)
	DeleteArtToysInCart(ctx context.Context, userId int64, artToyIds []int64) error
}
```

tbh, I don't think writing repository ahead is a good idea, I have literally no idea what these method is use for. Next time, please write after you know what service need to interact with database.

Reply:
I just write them from the tasks on linear😅

Reply:
ตรงไหนนะ